src/main/java/io/dropwizard/metrics5/WavefrontHistogram.java

```diff
@@ -0,0 +1,166 @@
+package io.dropwizard.metrics5;
+
+import com.wavefront.sdk.entities.histograms.WavefrontHistogramImpl;
+
+import java.io.OutputStream;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * WavefrontHistogram implementation
+ *
+ * @author Sushant Dewan (sushant@wavefront.com).
+ */
+public class WavefrontHistogram extends Histogram implements Metric {
+
+  private final WavefrontHistogramImpl delegate;
+
+  public static WavefrontHistogram get(MetricRegistry registry, MetricName metricName) {
+    return get(registry, metricName, System::currentTimeMillis);
+  }
+
+  public static WavefrontHistogram get(MetricRegistry registry,
+                                       MetricName metricName,
+                                       Supplier<Long> clock) {
+    // Awkward construction trying to fit in with Dropwizard Histogram
+    TDigestReservoir reservoir = new TDigestReservoir();
+    WavefrontHistogram tDigestHistogram = new WavefrontHistogram(reservoir, clock);
+    reservoir.set(tDigestHistogram);
+    try {
+      return registry.register(metricName, tDigestHistogram);
+    } catch(IllegalArgumentException e) {
+      Histogram existing = registry.histogram(metricName);
+      if (existing instanceof WavefrontHistogram) {
+        return (WavefrontHistogram) existing;
+      } else {
+        throw new IllegalStateException("Found non-WavefrontHistogram: " + existing);
```

Comment:
Probably just say, existing metric of type: <> found registered to MetricName: {} already.